### PR TITLE
feat(BUILD-4994): add link to repo

### PR DESCRIPTION
**Pull request by julien-carsique-sonarsource**
Add `repoURL` output.
Fix shellcheck warnings.
Add an `addLinkToPort` option to allow deactivating `add_link()` while the API overwrites the existing links instead of adding to them.

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -51,6 +51,9 @@ inputs:
     description: 'Whether to create a Port entity for the repository'
     required: false
     default: "true"
+outputs:
+  repoURL:
```
Comment (sebastienvermeille) [inline action.yaml:59]: I would rename this to camel case `repo-url` (evtl repoUrl) that way it follows a bit more what is commonly used in our code base such as here:
* https://github.com/SonarSource/gh-action_releasability/blob/master/action.yml#L5

And more commonly also used by GitHub:
https://github.com/SonarSource/gh-action_release/blob/master/.github/workflows/release.yml#L32

A pre-commit check for that would be great not sure how we could enforce it I will think about that :)

File: entrypoint.sh [hunk 2]
```diff
@@ -155,6 +156,8 @@ push_to_repository() {
     git branch -M master
     git remote add origin "https://oauth2:$github_token@github.com/$org_name/$repository_name.git"
     git push -u origin master
+    # add_link would be usable if not overwriting the existing links
+    # add_link "$repo_url"
```
Comment (sebastienvermeille) [inline entrypoint.sh]: I don't understand this comment.

A comment should explain **why** this code is written like that. (mostly because having to deal with an undesirable solution atm)

Maybe rephrase it like this: `# add_link can't be used here because existing links are overwritten ` ?

mino: this line should be removed: ` # add_link "$repo_url"` (if you really want to document how to call this function I would rather add it in the header of the function like that:

```bash
# add_link "$repo_url"
add_link()
```

File: entrypoint.sh [hunk 3]
```diff
@@ -155,6 +156,8 @@ push_to_repository() {
     git branch -M master
```
Comment (sebastienvermeille) [inline entrypoint.sh]: `master` is repeated twice here ~DRY principle

**Review by sebastienvermeille — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 2 above)
Comment (julien-carsique-sonarsource) [reply entrypoint.sh]: `add_link "$repo_url"` is pushing the repo URL to Port, which is nice. However, sadly this is overwriting the current URLs stored in Port. So we don't want to use it as it is working now.
I can make it an option instead.

File: action.yaml (hunk 1 above)
Comment (sebastienvermeille) [reply action.yaml:59]: edit same for freshly added `addLinkToPort` -> `add-link-to-port`

Comment (julien-carsique-sonarsource) [reply action.yaml:59]: I agree but I couldn't find a common pattern, and the purpose of camelCase here was for consistency with the inputs.

In port-actions, we use underscore, dash in releasability and sbom, and a mix of camelCase, dash and underscore in release action.

Comment (sebastienvermeille) [reply action.yaml:59]: ok so we keep doing a mix, ack.

**Review by sebastienvermeille — APPROVED** [review]
